cpeparse/rpmname.go

```diff
@@ -21,57 +21,73 @@ import (
 	"github.com/facebookincubator/nvdtools/wfn"
 )
 
+// split a string into two parts based on given index
+func split(s string, idx int) (left, right string) {
+	return s[:idx], s[idx+1:]
+}
+
 // FieldsFromRPMName returns name, version, release and acrhitecture parsed from RPM package name
-func FieldsFromRPMName(s string) (name, ver, rel, arch string, err error) {
-	pkg := s
+// NEVRA: https://blog.jasonantman.com/2014/07/how-yum-and-rpm-compare-versions/
+func FieldsFromRPMName(pkg string) (name, epoch, version, release, arch string, err error) {
+	// pkg should be name-(epoch:)version-release.arch.rpm
+
 	// extension
 	if strings.HasSuffix(pkg, ".rpm") {
 		pkg = pkg[:len(pkg)-4]
 	}
-	// architecture
-	if i := strings.LastIndexByte(pkg, '.'); i != -1 {
-		if arch, err = wfn.WFNize(pkg[i+1:]); err != nil {
-			err = fmt.Errorf("couldn't parse architecture from RPM package name %q: %v", s, err)
-			return
-		}
-		if arch == "noarch" || arch == "src" {
-			arch = wfn.Any
-		}
-		pkg = pkg[:i]
+
+	var i int
+
+	// name
+	if i = strings.IndexByte(pkg, '-'); i == -1 {
```

Comment:
Looks like this + `split` below can be done with a single call to `strings.SplitN()`